### PR TITLE
fix: default organizer bug in managed event type

### LINKED ISSUE #11920: [CAL-2623] Event Not Grabbing organizer's default app
https://threads.com/34542179472?s=iEVUQwvFgn5dEReTV3Yq85

https://discord.com/channels/1125390093386010654/1162108677633867866/1162108677633867866

<sub>[CAL-2623](https://linear.app/calcom/issue/CAL-2623/event-not-grabbing-organizers-default-app)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes  https://github.com/calcom/cal.com/issues/11920

TODO

- [x] Add E2E Test

BEFORE:- 
Screenshot 2023-10-16 at 8 03 48 PM

AFTER:-

Screenshot 2023-10-16 at 8 04 04 PM

How to reproduce?

1) Create a team and add a member
2) Create a managed event type for that team and assign it to other member.
3) Change the location to organizer default App
4) Book a meeting and check the location

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by CarinaWolli — COMMENTED** [review]
Location is missing on booking page, can we handle that in this PR too? 

Screenshot 2023-10-26 at 21 12 43

**Comment by CarinaWolli** [comment]
@Udit-takkar can you check why tests are failing? Not sure if it is related to your changes

**Review by CarinaWolli — DISMISSED** [review]
Looks good 👍

**Review by keithwillcode — DISMISSED** [review]
@Udit-takkar The E2E are failing consistently.

**Comment by Udit-takkar** [comment]
@keithwillcode checking

File: apps/web/playwright/managed-event-types.e2e.ts [hunk 1]
```diff
@@ -69,15 +71,36 @@ test.describe("Managed Event Types tests", () => {
       await page.goto("/event-types");
       await page.getByTestId("event-types").locator('a[title="managed"]').click();
       await page.getByTestId("vertical-tab-assignment").click();
-      await page.locator('[class$="control"]').filter({ hasText: "Select..." }).click();
+      await page.locator('[aria-label="assignment-dropdown"]').click();
```
Comment (ThyMinimalDev) [inline apps/web/playwright/managed-event-types.e2e.ts:74]: why not add a `data-testid` on the component and use `getByTestId`  instead

File: apps/web/components/eventtype/EventTeamTab.tsx [hunk 2]
```diff
@@ -76,6 +76,7 @@ const ChildrenEventTypesList = ({
       <div>
         <Label>{t("assign_to")}</Label>
         <ChildrenEventTypeSelect
+          aria-label="assignment-dropdown"
```
Comment (ThyMinimalDev) [inline apps/web/components/eventtype/EventTeamTab.tsx:79]: `data-testi`

File: apps/web/playwright/managed-event-types.e2e.ts [hunk 3]
```diff
@@ -69,15 +71,36 @@ test.describe("Managed Event Types tests", () => {
...
+    });
 
-      await adminUser.logout();
+    await test.step("Managed event type can use Organizer's default app as location", async () => {
+      await page.getByTestId("vertical-tab-event_setup_tab_title").click();
+      await expect(page.locator('[aria-label="Remove"]')).toBeVisible();
+      await page.locator('[aria-label="Remove"]').click();
+
+      await page.locator("#location-select").click();
+      await page.locator("text=Organizer's default app").click();
```
Comment (ThyMinimalDev) [inline apps/web/playwright/managed-event-types.e2e.ts:87]: I think we should get the text from the i18n using `localize` here

File: apps/web/playwright/managed-event-types.e2e.ts [hunk 4]
```diff
@@ -91,3 +114,9 @@ test.describe("Managed Event Types tests", () => {
     });
   });
 });
+
+async function gotoBookingPage(page: Page) {
+  const previewLink = await page.locator("[data-testid=preview-button]").getAttribute("href");
```
Comment (ThyMinimalDev) [inline apps/web/playwright/managed-event-types.e2e.ts:119]: `page.getByTestId`

File: apps/web/playwright/managed-event-types.e2e.ts (hunk 1 above)
Comment (Udit-takkar) [reply apps/web/playwright/managed-event-types.e2e.ts:74]: i tried passing data-testid to ReactSelect but i couldn't find it the dom tree

**Review by CarinaWolli — APPROVED** [review]
Looks good, @ThyMinimalDev's feedback has been addressed and tests are passing 👍

**Pull request merged**